### PR TITLE
Ssemenova/uniqueID

### PR DESCRIPTION
**Pull request by ssemenova**
### Description
 
[TNL-1593](https://openedx.atlassian.net/browse/TNL-1593)

Removed instances of duplicate IDs in ora

### Sandbox
- [ ] https://ssemenova.sandbox.edx.org/courses/course-v1:edx+ORA203+course/courseware/a4dfec19cf9b4a6fb5b18be6ccd9cecc/417e47b2663a4f79b62dba20b21628c8/

### Testing
- [ ] i18n
- [ ] RTL
- [ ] safecommit violation code review process
- [x] Unit, integration, acceptance tests as appropriate

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @cahrens 
- [x] Code review: @mushtaqak 
- [x] Code/a11y review: @cptvitamin 

### Post-review
- [ ] Squash commits

### CODE REVIEW

**Comment by cahrens** [comment]
@ssemenova  I found a non-unique ID. All I did was create 2 ORA problems on the same unit, publish, view in the LMS.

https://ssemenova.sandbox.edx.org/courses/course-v1:test+test+test/courseware/d472c272b6ec4fc59801797cd8a9d007/5e203085c57445729632397fdde87c2d/1?activate_block_id=block-v1%3Atest%2Btest%2Btest%2Btype%40vertical%2Bblock%4043a39cee1a6e45d29c790cbf4d184715

image

**Comment by cahrens** [comment]
Inconsistent styling on these collapsible regions (first two have very small icon, third is being styled as a link).

image

**Comment by cahrens** [comment]
@ssemenova this is how things look on prod (staging). So it looks like styling was lost for "Ideas" and "Content". I can't figure out how to enable "Provide Feedback on Peer Assessments". I think it might be something that isn't supported in Studio because the feature was never finished.

image

File: openassessment/templates/openassessmentblock/oa_rubric.html [hunk 1]
```diff
@@ -2,33 +2,31 @@
...
                 <div class="question__answers">
                     <div role="group" aria-labelledby="{{ rubric_type }}__assessment__rubric__prompt--{{ criterion.order_num }}">
                         {% for option in criterion.options %}
                         <div class="answer">
                             <div class="wrapper--input">
                                 <input type="radio"
                                        data-criterion-name="{{ criterion.name }}"
-                                       id="{{ rubric_type }}__assessment__rubric__question--{{ criterion.order_num }}__{{ option.order_num }}"
-                                       class="answer__value"
+                                       class="answer__value rubric_{{ criterion.order_num }}_{{ option.order_num }}"
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/oa_rubric.html]: Is this class value rubric_{{ criterion.order_num }}_{{ option.order_num }} actually needed (since there is also still an ID on the element)?

**Comment by ssemenova** [comment]
Right now all of those rubric-type headings are 14px. Should we make them larger than that?

**Comment by cahrens** [comment]
@ssemenova I'd suggest making them match what is on edx.org right now.

**Comment by cahrens** [comment]
These buttons have become quite large. They should be the same size as "Staff Debug Info".

image

**Comment by cahrens** [comment]
A couple of nits--

1) The first time you collapse the Staff Grade and Your Grade sections, they collapse immediately (without the animation). Subsequent expand/collapses have the animation.

2) The font on "Ideas", "Content", etc. buttons is not anti-aliased on production, which is why the font looks darker there. That's more an FYI than something to fix, as we are getting that styling from the fact that it is now a button.

File: openassessment/xblock/static/js/src/lms/oa_rubric.js [hunk 2]
```diff
@@ -214,7 +214,7 @@ OpenAssessment.Rubric.prototype = {
         var rubric = this;
         // Display appropriate messages for each selection
         $(selector, this.element).each(function(index, sel) {
-            var listItem = $(sel).parents(".assessment__rubric__question");
+            var listItem = $(sel).parents("#assessment__rubric__question");
```
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_rubric.js]: Is this (the switch from a class to ID) correct?

File: test/acceptance/accessibility.py [hunk 3]
```diff
@@ -253,32 +253,21 @@ def setUp(self):
...
 
     def test_multiple_ora_complete_flow(self):
         """
         Test accessibility when we have a unit containing multiple ORA blocks.
         """
+        # Assess entire problem page for potential duplicate IDs among ORA xblocks
+        self.multiple_assessment_page.a11y_audit.config.set_scope(['#main'])
+        self.multiple_assessment_page.a11y_audit.config.set_rules({
+            "apply": [
+                "duplicate-ID"
```
Comment (cahrens) [inline test/acceptance/accessibility.py]: This is normally lower-case. Make sure it is running!

File: test/acceptance/pages.py [hunk 4]
```diff
@@ -15,6 +15,34 @@ class PageConfigurationError(Exception):
     pass
 
 
+class MultipleAssessmentPage(PageObject):
```
Comment (cahrens) [inline test/acceptance/pages.py]: BaseAssessmentPage

Remove is_browser_on_page

Have MultipleAssessmentPage extend BaseAssessmentPage and add is_browser_on_page.

Also have OpenAssessmentPage extend BaseAssessmentPage.

File: test/acceptance/pages.py [hunk 5]
```diff
@@ -15,6 +15,34 @@ class PageConfigurationError(Exception):
...
+        Args:
+            browser (Selenium browser): The browser object used by the tests.
+            problem_location (unicode): URL path for the problem, appended to the base URL.
+
+        """
+        super(MultipleAssessmentPage, self).__init__(browser)
+        self._problem_location = problem_location
+
+    def is_browser_on_page(self):
+        return self.q(css='#main').is_present()
```
Comment (cahrens) [inline test/acceptance/pages.py]: Add a comment why the css is #main (don't want to limit to a specific ORA problem).

File: openassessment/xblock/static/js/src/lms/oa_rubric.js (hunk 2 above)
Comment (ssemenova) [inline openassessment/xblock/static/js/src/lms/oa_rubric.js]: Yep

File: openassessment/xblock/static/sass/xb/elements/_controls.scss [hunk 6]
```diff
@@ -108,7 +108,6 @@
 
 %btn-reset {
```
Comment (cahrens) [inline openassessment/xblock/static/sass/xb/elements/_controls.scss]: I am worried about potential fallout. Can you style just the things that aren't really buttons with the size 14?

File: openassessment/xblock/static/sass/oa/utilities/_developer.scss [hunk 7]
```diff
@@ -47,6 +47,7 @@ $link-hover: $edx-blue-l1 !default; // from our Pattern Library http://ux.edx.or
       @extend %btn-reset;
 
       padding: ($baseline-v/4) ($baseline-v/2);
+      font-size: 12px;
```
Comment (cahrens) [inline openassessment/xblock/static/sass/oa/utilities/_developer.scss]: Same note as previous (switch which ones gets special handling).

File: openassessment/xblock/static/js/src/lms/oa_rubric.js (hunk 2 above)
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_rubric.js]: How does assessment__rubric__question end up being unique then?

**Comment by cahrens** [comment]
@mushtaqak Could you provide a second code review of this PR as it finishes up the work you began in #894?

File: openassessment/xblock/static/sass/oa/views/_oa-base.scss [hunk 8]
```diff
@@ -430,7 +430,7 @@
   // --------------------
   .ui-slidable__container {
     .ui-slidable__content {
-      display: none;
+      //display: none;
```
Comment (cahrens) [inline openassessment/xblock/static/sass/oa/views/_oa-base.scss]: Remove this commented-out line.

File: openassessment/xblock/static/sass/oa/views/_oa-base.scss [hunk 9]
```diff
@@ -441,9 +441,15 @@
...
+    &.is--initially--collapsed  {
+      .ui-slidable__content {
+        display: none;
+      }
+    }
+
     &.is--showing {
       .ui-slidable__content {
-        display: block;
+        //display: block;
```
Comment (cahrens) [inline openassessment/xblock/static/sass/oa/views/_oa-base.scss]: And this too.

**Comment by cahrens** [comment]
Clicking on the "Review your ..." link scrolls the page down, but does not actually move keyboard focus.

image

**Comment by cahrens** [comment]
In the student training view, I was surprised that this "disabled" button is focusable and clickable (via the keyboard). 

image

File: openassessment/templates/openassessmentblock/student_training/student_training_error.html [hunk 10]
```diff
@@ -1,8 +1,13 @@
...
 {% load i18n %}
 
+{% block list_item %}
+<li class="openassessment__steps__step step--student-training is--showing ui-slidable__container is--initially--collapsed {% if allow_latex %}allow--latex{%endif%}"
+    tabindex="-1">
+{% endblock %}
+
 {% block button %}
-    <button class="ui-slidable" aria-expanded="false" disabled="disabled" id="oa_training_{{ xblock_id }}" aria-controls="oa_training_{{ xblock_id }}_content">
+    <button class="ui-slidable" aria-expanded="false" id="oa_training_{{ xblock_id }}" aria-controls="oa_training_{{ xblock_id }}_content">
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/student_training/student_training_error.html]: Why did you remove the disabled here?

File: openassessment/xblock/static/js/src/lms/oa_base.js [hunk 11]
```diff
@@ -96,6 +96,10 @@ OpenAssessment.BaseView.prototype = {
                     $toggleButton.attr('aria-expanded', 'true');
                     $container.addClass('is--showing');
                 }
+
+                if ($container.hasClass('is--initially--collapsed ')) {
+                    $container.removeClass('is--initially--collapsed ');
```
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_base.js]: I don't think there is any harm in just always calling removeClass, regardless of whether or not is--initially--collapsed is present. it will be a no-op if the class is not there.

**Comment by cahrens** [comment]
In the "Past Assignment" example, the step names should still be styled like buttons (blue), as they are in production.

Sandbox:
image


Production:
image

File: openassessment/templates/openassessmentblock/student_training/student_training_error.html (hunk 10 above)
Comment (ssemenova) [inline openassessment/templates/openassessmentblock/student_training/student_training_error.html]: It has a body block, so it should be expandable - I don't think I caught it the first time I was making the buttons disabled.

**Comment by ssemenova** [comment]
@cahrens 

The "review your..." link and self assessment links:
The review your link was actually going to the wrong ID but the self assessment seems to be right. They're supposed to go to the div encompassing the step, so when you tab it tabs into the button. 

The disabled bug seems like an ok fix. They should have disabled="disabled" attributes and the js is already changing the is--disabled class so it shouldn't be hard to add the disabled attribute.

File: openassessment/xblock/static/js/src/lms/oa_rubric.js (hunk 2 above)
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_rubric.js]: @ssemenova ^

**Comment by cahrens** [comment]
@ssemenova Done re-reviewing. Let's leave the "disabled" button issue for a future story, as it's pre-existing and pervasive. I created a bug for it.

File: openassessment/xblock/openassessmentblock.py [hunk 12]
```diff
@@ -45,37 +45,37 @@
 UI_MODELS = {
     "submission": {
         "name": "submission",
-        "class_id": "openassessment__response",
+        "class_id": "step--response",
```
Comment (mushtaqak) [inline openassessment/xblock/openassessmentblock.py]: Nice work :)

**Comment by mushtaqak** [comment]
Sorry for reviewing late.

@ssemenova This looks great. Nice clean up 👍

**Comment by cahrens** [comment]
👍 @ssemenova this is good to merge, and let's do an ORA release! (by which I mean, you can do an ORA release!).
